src/daemon/daemon.cpp

```diff
@@ -364,24 +366,54 @@ auto try_mem_size(const std::string& val) -> mp::optional<mp::MemorySize>
 }
 
 std::vector<mp::NetworkInterface> validate_extra_interfaces(const mp::LaunchRequest* request,
+                                                            const mp::VirtualMachineFactory& factory,
                                                             mp::LaunchError& option_errors)
 {
-    // The associated bool indicates whether the interface mode is 'auto', i.e., needs to be configured by cloud-init.
+    std::vector<mp::NetworkInterfaceInfo> factory_networks;
+
+    if (!request->network_options().empty())
+    {
+        try
+        {
+            factory_networks = factory.list_networks();
+        }
+        catch (const mp::NotImplementedOnThisBackendException&)
+        {
+            // If list-networks is not implemented, we should report that --network is not implemented on this backend.
+            throw mp::NotImplementedOnThisBackendException("--network");
+        }
+    }
+
     std::vector<mp::NetworkInterface> interfaces;
 
     for (const auto& net : request->network_options())
     {
+        // Check that the id the user specified is valid.
+        auto pred = [net](const mp::NetworkInterfaceInfo& info) { return info.id == net.id(); };
+        const auto& result = std::find_if(factory_networks.cbegin(), factory_networks.cend(), pred);
+
+        if (result == factory_networks.cend())
+        {
+            mpl::log(mpl::Level::warning, category, fmt::format("Invalid network id \"{}\"", net.id()));
+            option_errors.add_error_codes(mp::LaunchError::INVALID_NETWORK);
+        }
+
+        // Check that the MAC the user specified is not repeated.
```

Comment:
Humm, that's not what this is doing, that happens elsewhere. This is just validating syntax.

Reply:
Oops, this comment must be a leftover of some previous implementation. Corrected.

Reply:
Ah, because it was copy pasted from below? There's a case that tying the comment to the code would have prevented :wink:

Reply:
You are indisputably right :S